yq.go

```diff
@@ -241,6 +242,41 @@ Note that if you set both flags only overwrite will take effect.
 	return cmdMerge
 }
 
+func createLintCmd() *cobra.Command {
+	var cmdLint = &cobra.Command{
+		Use:     "lint [yaml_file]",
+		Aliases: []string{"l"},
+		Short:   "yq l sample.yaml",
+		Example: `
+yq lint sample.yaml
+yq l sample.yaml
+      `,
+		Long: "Does a basic lint on the file to see if it can be parsed",
+		RunE: lint,
+	}
+	return cmdLint
+}
+
+func lint(cmd *cobra.Command, args []string) error {
+	var path = ""
+	if len(args) < 1 {
+		return errors.New("Must provide filename")
+	} else if len(args) >= 1 {
+		path = args[0]
+	}
+	obj := make(map[string]interface{})
+	bytes, err := ioutil.ReadFile(path)
+	if err != nil {
+		return err
+	}
+	err = yaml.Unmarshal(bytes, &obj)
+	if err != nil {
+		return errors.New("Failed lint")
```

Comment:
why not return the actual error which should provide information about why the yaml file is not valid?

Reply:
Guessing this will go away with using `readProperty` and capturing errors